src/Sylius/Bundle/OrderBundle/NumberGenerator/SequentialOrderNumberGenerator.php

```diff
@@ -64,11 +76,30 @@ public function __construct(
      */
     public function generate(OrderInterface $order)
     {
-        $sequence = $this->getSequence();
-        
-        $number = $this->generateNumber($sequence->getIndex());
-        $sequence->incrementIndex();
-        
+        $number = null;
+        $counter = 0;
+        $done = false;
+        while (!$done) {
+            if (++$counter > 100) {
```

Comment:
Why do you need this `$counter` i don't get it. I suppose that is some kind of mechanism to break the loop but still i don't get it. It looks like `sleep`. Maybe something like this will be more readable:
```php
$counter = 0;
$done = false;
do {
stuff....

} while (!$done || ++$counter < 100);
```